### PR TITLE
feat: live printing bed mesh changes

### PR DESCRIPTION
**Pull request by pedrolamas**
This a considerable refactor of Bed Mesh in Fluidd and will need further testing, but brings the following major changes:

- Bed mesh profiles are now retrieved from `printer.bed_mesh.profiles` object instead of config settings (reason being that this object is "live" updated, but the config never changes)
- Bed mesh profiles can be loaded/cleared/changed while printer is printing (I see no reason to not allow this if Klipper allows for it), calibration and homing will still be disabled if printer is busy.
- Bed mesh save/delete will NOT restart the printer (it will instead show the "SAVE CONFIG AND RESTART" button on top and wait for user confirmation - all translations have been updated accordingly!

Note: this might be easier to review with "hide whitespace" option [enabled](1056/files?w=1)!

### CODE REVIEW

**Review by matmen — COMMENTED** [review]
LGTM, but are the buttons (clear profile, delete profile) enabled while printing intentionally now? Not sure what the side effects of deleting a profile while printing would be..

**Comment by pedrolamas** [comment]
All good points, I've been giving some though on this for the last couple of days!

Starting on delete, once a bed mesh is loaded into memory, the profile can be deleted and the mesh data will remain in memory, so deleting a profile while printing has zero impact on the active print.

Now loading or clearing a profile while printing will impact the active print. I agree that a simple click as it currently is might not be appropriate... so maybe show a confirmation message for clear/load if state printing?

File: src/components/widgets/bedmesh/BedMeshControls.vue [hunk 1]
```diff
@@ -365,7 +359,7 @@ export default class BedMesh extends Mixins(StateMixin, ToolheadMixin) {
 
   // If we have a default profile.
   get hasDefault (): boolean {
-    return (this.meshes.findIndex(mesh => mesh.profile_name === 'default') > -1)
+    return (this.bedMeshProfiles.findIndex(mesh => mesh.name === 'default') > -1)
   }
```
[the remark this answers is not in the document]
Comment (pedrolamas) [reply src/components/widgets/bedmesh/BedMeshControls.vue]: For some weird reason GitHub didn't allow me to merge this change, so I made it manually on a separate commit.

**Comment by pedrolamas** [comment]
I've now added confirmation dialogs for mesh load/clear if printer printing (no confirmation dialog is shown if printer is paused, I see no reason for that as there is no big impact anyway).

image

**Review by matmen — APPROVED** [review]
LGTM!

**Pull request merged**